.github/scripts/run-librispeech-pruned-transducer-stateless7-2022-11-11.sh

```diff
@@ -30,6 +30,14 @@ ln -s pretrained.pt epoch-99.pt
 ls -lh *.pt
 popd
 
+log "Test exporting to ONNX format"
+./pruned_transducer_stateless7/export.py \
+  --exp-dir $repo/exp \
+  --bpe-model $repo/data/lang_bpe_500/bpe.model \
+  --epoch 99 \
```

Comment:
Do you need to use `--use-averaged-model=false`? Otherwise, I am afraid it would fail.

Reply:
Could you also modifiy
https://github.com/k2-fsa/icefall/blob/65d7192dca03ba21bff4270add3891c9730491a7/.github/workflows/run-librispeech-2022-11-11-stateless7.yml#L42

You need to add another condition
```
github.event.label.name == 'onnx'
```

Reply:
The problem is that ı am separating onnx parts with torch.onnx.is_in_onnx_export(). However jit cannot export with this function. I opened a [topic in Pytorch Discuss](https://discuss.pytorch.org/t/exporting-onnx-and-jit-model-at-the-same-time/168759)

Reply:
Could you change
https://github.com/k2-fsa/icefall/blob/d40e8b9f25adea99ff98d854b01211f56caaacc2/egs/librispeech/ASR/pruned_transducer_stateless7/scaling.py#L301-L302
to
```python3
 if torch.jit.is_scripting(): 
     return x.softmax(dim)
elif torch.onnx.is_in_onnx_export():
     return x.softmax(dim)
```

Reply:
I can change these lines, but I cannot find a workaround for here
https://github.com/k2-fsa/icefall/blob/d40e8b9f25adea99ff98d854b01211f56caaacc2/egs/librispeech/ASR/pruned_transducer_stateless7/zipformer.py#L1261-L1280

Reply:
and here
https://github.com/k2-fsa/icefall/blob/d40e8b9f25adea99ff98d854b01211f56caaacc2/egs/librispeech/ASR/pruned_transducer_stateless7/zipformer.py#L1471-L1475

Reply:
What is the error log for
https://github.com/k2-fsa/icefall/blob/d40e8b9f25adea99ff98d854b01211f56caaacc2/egs/librispeech/ASR/pruned_transducer_stateless7/zipformer.py#L1261-L1280

I think our current code is using
https://github.com/k2-fsa/icefall/blob/d40e8b9f25adea99ff98d854b01211f56caaacc2/egs/librispeech/ASR/pruned_transducer_stateless2/conformer.py#L1012
instead of `torch.onnx.is_in_onnx_export()`.

Reply:
Error log is 
```
(k2sh) yunusemre.ozkose@server1:/path/to/contribution/v13_icefall/egs/librispeech/ASR$ ./pruned_transducer_stateless7/export.py \
>   --exp-dir /path/to/models/icefall-asr-librispeech-pruned-transducer-stateless7-2022-11-11/exp_ \
>   --bpe-model /path/to/models/icefall-asr-librispeech-pruned-transducer-stateless7-2022-11-11/exp_/bpe.model \
>   --epoch 30 \
>   --avg 1 \
>   --use-averaged-model False \
>   --jit 1
2022-12-22 08:23:22,242 INFO [export.py:438] device: cuda:0
2022-12-22 08:23:22,259 INFO [export.py:447] {'best_train_loss': inf, 'best_valid_loss': inf, 'best_train_epoch': -1, 'best_valid_epoch': -1, 'batch_idx_train': 0, 'log_interval': 50, 'reset_interval': 200, 'valid_interval': 3000, 'feature_dim': 80, 'subsampling_factor': 4, 'warm_step': 2000, 'env_info': {'k2-version': '1.21', 'k2-build-type': 'Release', 'k2-with-cuda': True, 'k2-git-sha1': 'c3a7404e0d9da4d8d99f3f15c54d80b321c1b960', 'k2-git-date': 'Mon Oct 31 07:45:41 2022', 'lhotse-version': '1.10.0.dev+git.811db9a.clean', 'torch-version': '1.10.0', 'torch-cuda-available': True, 'torch-cuda-version': '11.3', 'python-version': '3.8', 'icefall-git-branch': 'zipformer_onnx_export', 'icefall-git-sha1': 'd40e8b9-dirty', 'icefall-git-date': 'Tue Dec 20 12:15:01 2022', 'icefall-path': '/path/to/icefall_221125', 'k2-path': '/home/yunusemre.ozkose/anaconda3/envs/k2sh/lib/python3.8/site-packages/k2-1.21.dev20221103+cuda11.3.torch1.10.0-py3.8-linux-x86_64.egg/k2/__init__.py', 'lhotse-path': '/home/yunusemre.ozkose/anaconda3/envs/k2sh/lib/python3.8/site-packages/lhotse/__init__.py', 'hostname': 'boxx-4', 'IP address': '192.168.10.204'}, 'epoch': 30, 'iter': 0, 'avg': 1, 'use_averaged_model': False, 'exp_dir': PosixPath('/path/to/models/icefall-asr-librispeech-pruned-transducer-stateless7-2022-11-11/exp_'), 'bpe_model': '/path/to/models/icefall-asr-librispeech-pruned-transducer-stateless7-2022-11-11/exp_/bpe.model', 'jit': True, 'onnx': False, 'context_size': 2, 'num_encoder_layers': '2,4,3,2,4', 'feedforward_dims': '1024,1024,2048,2048,1024', 'nhead': '8,8,8,8,8', 'encoder_dims': '384,384,384,384,384', 'attention_dims': '192,192,192,192,192', 'encoder_unmasked_dims': '256,256,256,256,256', 'zipformer_downsampling_factors': '1,2,4,8,2', 'cnn_module_kernels': '31,31,31,31,31', 'decoder_dim': 512, 'joiner_dim': 512, 'blank_id': 0, 'vocab_size': 500}
2022-12-22 08:23:22,259 INFO [export.py:449] About to create model
2022-12-22 08:23:22,624 INFO [zipformer.py:178] At encoder stack 4, which has downsampling_factor=2, we will combine the outputs of layers 1 and 3, with downsampling_factors=2 and 8.
2022-12-22 08:23:27,572 INFO [checkpoint.py:112] Loading checkpoint from /path/to/models/icefall-asr-librispeech-pruned-transducer-stateless7-2022-11-11/exp_/epoch-30.pt
2022-12-22 08:23:37,067 INFO [export.py:565] Using torch.jit.script
Traceback (most recent call last):
  File "./pruned_transducer_stateless7/export.py", line 583, in <module>
    main()
  File "/home/yunusemre.ozkose/anaconda3/envs/k2sh/lib/python3.8/site-packages/torch/autograd/grad_mode.py", line 28, in decorate_context
    return func(*args, **kwargs)
  File "./pruned_transducer_stateless7/export.py", line 566, in main
    model = torch.jit.script(model)
  File "/home/yunusemre.ozkose/anaconda3/envs/k2sh/lib/python3.8/site-packages/torch/jit/_script.py", line 1257, in script
    return torch.jit._recursive.create_script_module(
  File "/home/yunusemre.ozkose/anaconda3/envs/k2sh/lib/python3.8/site-packages/torch/jit/_recursive.py", line 451, in create_script_module
    return create_script_module_impl(nn_module, concrete_type, stubs_fn)
  File "/home/yunusemre.ozkose/anaconda3/envs/k2sh/lib/python3.8/site-packages/torch/jit/_recursive.py", line 513, in create_script_module_impl
    script_module = torch.jit.RecursiveScriptModule._construct(cpp_module, init_fn)
  File "/home/yunusemre.ozkose/anaconda3/envs/k2sh/lib/python3.8/site-packages/torch/jit/_script.py", line 587, in _construct
    init_fn(script_module)
  File "/home/yunusemre.ozkose/anaconda3/envs/k2sh/lib/python3.8/site-packages/torch/jit/_recursive.py", line 491, in init_fn
    scripted = create_script_module_impl(orig_value, sub_concrete_type, stubs_fn)
  File "/home/yunusemre.ozkose/anaconda3/envs/k2sh/lib/python3.8/site-packages/torch/jit/_recursive.py", line 513, in create_script_module_impl
    script_module = torch.jit.RecursiveScriptModule._construct(cpp_module, init_fn)
  File "/home/yunusemre.ozkose/anaconda3/envs/k2sh/lib/python3.8/site-packages/torch/jit/_script.py", line 587, in _construct
    init_fn(script_module)
  File "/home/yunusemre.ozkose/anaconda3/envs/k2sh/lib/python3.8/site-packages/torch/jit/_recursive.py", line 491, in init_fn
    scripted = create_script_module_impl(orig_value, sub_concrete_type, stubs_fn)
  File "/home/yunusemre.ozkose/anaconda3/envs/k2sh/lib/python3.8/site-packages/torch/jit/_recursive.py", line 513, in create_script_module_impl
    script_module = torch.jit.RecursiveScriptModule._construct(cpp_module, init_fn)
  File "/home/yunusemre.ozkose/anaconda3/envs/k2sh/lib/python3.8/site-packages/torch/jit/_script.py", line 587, in _construct
    init_fn(script_module)
  File "/home/yunusemre.ozkose/anaconda3/envs/k2sh/lib/python3.8/site-packages/torch/jit/_recursive.py", line 491, in init_fn
    scripted = create_script_module_impl(orig_value, sub_concrete_type, stubs_fn)
  File "/home/yunusemre.ozkose/anaconda3/envs/k2sh/lib/python3.8/site-packages/torch/jit/_recursive.py", line 513, in create_script_module_impl
    script_module = torch.jit.RecursiveScriptModule._construct(cpp_module, init_fn)
  File "/home/yunusemre.ozkose/anaconda3/envs/k2sh/lib/python3.8/site-packages/torch/jit/_script.py", line 587, in _construct
    init_fn(script_module)
  File "/home/yunusemre.ozkose/anaconda3/envs/k2sh/lib/python3.8/site-packages/torch/jit/_recursive.py", line 491, in init_fn
    scripted = create_script_module_impl(orig_value, sub_concrete_type, stubs_fn)
  File "/home/yunusemre.ozkose/anaconda3/envs/k2sh/lib/python3.8/site-packages/torch/jit/_recursive.py", line 513, in create_script_module_impl
    script_module = torch.jit.RecursiveScriptModule._construct(cpp_module, init_fn)
  File "/home/yunusemre.ozkose/anaconda3/envs/k2sh/lib/python3.8/site-packages/torch/jit/_script.py", line 587, in _construct
    init_fn(script_module)
  File "/home/yunusemre.ozkose/anaconda3/envs/k2sh/lib/python3.8/site-packages/torch/jit/_recursive.py", line 491, in init_fn
    scripted = create_script_module_impl(orig_value, sub_concrete_type, stubs_fn)
  File "/home/yunusemre.ozkose/anaconda3/envs/k2sh/lib/python3.8/site-packages/torch/jit/_recursive.py", line 513, in create_script_module_impl
    script_module = torch.jit.RecursiveScriptModule._construct(cpp_module, init_fn)
  File "/home/yunusemre.ozkose/anaconda3/envs/k2sh/lib/python3.8/site-packages/torch/jit/_script.py", line 587, in _construct
    init_fn(script_module)
  File "/home/yunusemre.ozkose/anaconda3/envs/k2sh/lib/python3.8/site-packages/torch/jit/_recursive.py", line 491, in init_fn
    scripted = create_script_module_impl(orig_value, sub_concrete_type, stubs_fn)
  File "/home/yunusemre.ozkose/anaconda3/envs/k2sh/lib/python3.8/site-packages/torch/jit/_recursive.py", line 517, in create_script_module_impl
    create_methods_and_properties_from_stubs(concrete_type, method_stubs, property_stubs)
  File "/home/yunusemre.ozkose/anaconda3/envs/k2sh/lib/python3.8/site-packages/torch/jit/_recursive.py", line 368, in create_methods_and_properties_from_stubs
    concrete_type._create_methods_and_properties(property_defs, property_rcbs, method_defs, method_rcbs, method_defaults)
  File "/home/yunusemre.ozkose/anaconda3/envs/k2sh/lib/python3.8/site-packages/torch/jit/_recursive.py", line 838, in try_compile_fn
    return torch.jit.script(fn, _rcb=rcb)
  File "/home/yunusemre.ozkose/anaconda3/envs/k2sh/lib/python3.8/site-packages/torch/jit/_script.py", line 1307, in script
    ast = get_jit_def(obj, obj.__name__)
  File "/home/yunusemre.ozkose/anaconda3/envs/k2sh/lib/python3.8/site-packages/torch/jit/frontend.py", line 264, in get_jit_def
    return build_def(parsed_def.ctx, fn_def, type_line, def_name, self_name=self_name, pdt_arg_types=pdt_arg_types)
  File "/home/yunusemre.ozkose/anaconda3/envs/k2sh/lib/python3.8/site-packages/torch/jit/frontend.py", line 315, in build_def
    build_stmts(ctx, body))
  File "/home/yunusemre.ozkose/anaconda3/envs/k2sh/lib/python3.8/site-packages/torch/jit/frontend.py", line 137, in build_stmts
    stmts = [build_stmt(ctx, s) for s in stmts]
  File "/home/yunusemre.ozkose/anaconda3/envs/k2sh/lib/python3.8/site-packages/torch/jit/frontend.py", line 137, in <listcomp>
    stmts = [build_stmt(ctx, s) for s in stmts]
  File "/home/yunusemre.ozkose/anaconda3/envs/k2sh/lib/python3.8/site-packages/torch/jit/frontend.py", line 286, in __call__
    raise UnsupportedNodeError(ctx, node)
torch.jit.frontend.UnsupportedNodeError: import statements aren't supported:
  File "/home/yunusemre.ozkose/anaconda3/envs/k2sh/lib/python3.8/site-packages/torch/onnx/__init__.py", line 386
    """

    from torch.onnx import utils
    ~~~~ <--- HERE
    return utils.is_in_onnx_export()
```

Reply:
During onnx export, `is_jit_tracing()` become False. It can be related to torch version, I am checking.

Reply:
I can export pruned_transducer_stateless3, but cannot export pruned_transducer_stateless7. So , it is not related to env. I am checking differences.

Reply:
`torch.onnx.export` by default uses `torch.jit.trace()`, so I suggest that you delete `torch.onnx.is_in_onnx_export`.

Reply:
I just release that I am using `torch.jit.is_scripting()` instead of `torch.jit.is_tracing()` :)